src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java

```diff
@@ -297,11 +301,17 @@ public static int getGeometryType(Geometry geometry) {
     public GeoJSONBuilder value(Object value) {
         if (value instanceof java.util.Date || value instanceof Calendar) {
             value = Converters.convert(value, String.class);
+        } else if (value instanceof Double) {
+            value = roundToNumberOfDecimalPlaces((Double)value);
         }
         super.value(value);
         return this;
     }
-    
+
+    private double roundToNumberOfDecimalPlaces(Double value) {
+        return new BigDecimal((Double)value).setScale(numDecimals, BigDecimal.ROUND_HALF_UP).doubleValue();		
```

Comment:
For best performance, we could also dodge the jsonlib creation of a Double wrapper, override value(double d) and make it call directly append(formattedStringValue) instead.

As a separate observation, your change has the side effect of also formatting attributes that happen to be doubles, which we might need to preserve with full accuracy. Probably best to change writeCoordinate to call onto an optimized writeOrdinate method that formats the double the efficient way, and then appends a string